perf/roles/trex/tasks/main.yml

```diff
@@ -1,37 +1,58 @@
 ---
 
-- name: install build dependencies
+- name: install runtime dependencies
   dnf:
     name:
-      - gcc
-      - gcc-c++
-      - git
       - python3
-      - rdma-core-devel
-      - zlib-devel
+      - ca-certificates
+      - python3-cryptography
     state: installed
   tags: install
 
+- name: Update CA certificates
+  command: update-ca-trust enable
+
+- name: Generate private key
+  community.crypto.openssl_privatekey:
+    path: /etc/pki/tls/private/private.key
+    size: 2048
+  register: private_key_result
+
+- name: Generate an OpenSSL Certificate Signing Request
+  community.crypto.openssl_csr:
+    path: /etc/pki/tls/private/cert.csr
+    privatekey_path: /etc/pki/tls/private/private.key
+    common_name: www.ansible.com
+
+- name: Generate a Self Signed OpenSSL certificate
+  community.crypto.x509_certificate:
+    path: /etc/pki/ca-trust/source/anchors/certificate.crt
+    privatekey_path: /etc/pki/tls/private/private.key
+    csr_path: /etc/pki/tls/private/cert.csr
+    provider: selfsigned
+
+- name: Extract and update trusted CA's
+  command: update-ca-trust extract
+
 - name: stop existing trex server
   command: pkill -x _t-rex-64
   ignore_errors: true
   tags: setup
 
-- name: clone trex
-  git:
-    repo: '{{trex_repo}}'
-    dest: '{{trex_dir}}'
-    version: '{{trex_tag}}'
-    force: true
+- name: Create TRex install directory
+  file:
+    path: "{{ trex_install_dir }}"
+    state: directory
+    mode: "0755"
   tags: install
 
-- name: configure and build trex
-  command:
-    cmd: './b {{item}}'
-    chdir: '{{trex_dir}}/linux_dpdk'
-  loop:
-    - configure
-    - build
+- name: Download and Extract TRex release
+  unarchive:
+    src: "{{ trex_release_url }}/{{ trex_version }}.tar.gz"
+    dest: "{{ trex_install_dir }}"
+    remote_src: yes
+    extra_opts: --strip-components=1
+    validate_certs: no
```

Comment:
And you are still ignoring invalid SSL certificates. What I was suggesting in my last review is to import the Certificate Authority that has signed https://trex-tgn.cisco.com/trex/release/ certificate. This CA is not present in the standard certs bundles but seems to be known by web browsers.

Let me know if you need some help for this.

Reply:
Removed self signed certificate and ignoring the certificate check ...